articles/users/search.md

```diff
@@ -27,6 +28,25 @@ In this document, we use the terms **eventually consistent** and **immediately c
 
 * **Immediately consistent**: When you request information about a user (or a group of users), the response will reflect the results of all successful write operations, including those that occured shortly prior to your request.
 
+## General Principles
+
+When running user searches:
+
+* Use an immediately consistent endpoint during authentication pipeline
+* Use exact match searches (with the `raw` subfield) whenever possible
+* Use a well-known schema for metadata:
+  * Use consistent data types for properties
+  * Avoid dynamic property names
+  * Avoid large schema sizes and deep structures
+  * Avoid storing data you do not need for authentication and authorization purposes
+
+Please **avoid**:
```

Comment:
Lets remove this `Please avoid` section. If the product supports this it looks weird we ask the user to "avoid" using.